### PR TITLE
Integrated mocha and chai for backend testing purposes.

### PR DESCRIPTION
**Pull request by recursion**
Basic mocha integration with a new grunt task for running server tests. This changes nothing as far as jasmine is concerned. In fact it changes nothing at all,  except that there is now a grunt task for running backend tests using mocha and chai.

- Added grunt task 'test:server' for running files using the format <*.spec.js> in the specs dir 'test/server/'. - Included a very simple user.model.specs.js file as demo.

Long story short - I  ran into some issues that made me install mocha/chai.  Particularly afterAll()/beforeAll and done().

### CODE REVIEW

**Comment by mnaamani** [comment]
Nice, I think this should encourage us to write tests without any hassle!
would we want to maybe .gitignore the results.txt which seems to be generated by tests?
